tools/scripts/features/frcnn/extract_features_frcnn.py

```diff
@@ -0,0 +1,256 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import argparse
+import copy
+import glob
+import os
+
+import numpy as np
+import torch
+
+from modeling_frcnn import GeneralizedRCNN
+from processing_image import Preprocess
+from utils import Config
+
+
+class FeatureExtractor:
+
+    MODEL_URL = {"LXMERT": "unc-nlp/frcnn-vg-finetuned"}
+
+    def __init__(self):
+        self.args = self.get_parser().parse_args()
+        self.frcnn, self.frcnn_cfg = self._build_detection_model()
+
+    def get_parser(self):
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "--model_name",
+            default="LXMERT",
+            type=str,
+            help="Model to use for detection",
+        )
+        parser.add_argument(
+            "--model_file",
+            default=None,
+            type=str,
+            help="Huggingface model file. This overrides the model_name param.",
+        )
+        parser.add_argument(
+            "--config_file", default=None, type=str, help="Huggingface config file"
+        )
+        parser.add_argument(
+            "--start_index", default=0, type=int, help="Index to start from "
+        )
+        parser.add_argument("--end_index", default=None, type=int, help="")
+        parser.add_argument("--batch_size", type=int, default=2, help="Batch size")
+        parser.add_argument(
+            "--output_folder", type=str, default="./output", help="Output folder"
+        )
+        parser.add_argument("--image_dir", type=str, help="Image directory or file")
+        parser.add_argument(
+            "--feature_index",
```

Comment:
Is this the feature type "fc6" or "fc7" etc?

Reply:
This is what I replaced feature type with, I guess I did not entirely understand what feature type represented so I replaced it with what I assumed it meant.

This takes the index of the feature we are trying to extract (71 is for bike given our current model, etc.)

Reply:
Feature type is not equivalent to object type. Features can be of two types depending on from which layer of the FRCNN model they are extracted. If it is the last layer then it is fc7 and if it is from the previous layer it is fc6.